### PR TITLE
Travis CI: change to docker infrastructure.

### PR DESCRIPTION
**Pull request by antiagainst**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,27 +26,28 @@ branches:
...
     - master
 
-install:
-  # Travis CI is powered with Ubuntu 12.04, whose toolchain is outdated.
-  - sudo add-apt-repository --yes ppa:smspillaz/cmake-2.8.12
-  - sudo add-apt-repository --yes ppa:ubuntu-toolchain-r/test
-  - if [ "$CC" = "clang" ]; then wget -O - http://llvm.org/apt/llvm-snapshot.gpg.key | sudo apt-key add -; fi
-  - if [ "$CC" = "clang" ]; then sudo add-apt-repository --yes 'deb http://llvm.org/apt/precise/ llvm-toolchain-precise-3.6 main'; fi
-  - sudo apt-get update -qq
+# Travis CI is powered with Ubuntu 12.04, whose toolchain is outdated.
```
Comment (dmircevski) [inline .travis.yml]: Is this still relevant?  Would any of the lines below be unnecessary on a different Ubuntu version?

Comment (antiagainst) [inline .travis.yml]: The following configuration is only for Ubuntu 12.04. If Travis updated it OS, the following will be unnecessary.

**Comment by dmircevski** [comment]
+2

**Comment by dneto0** [comment]
+2

**Pull request merged**